frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -140,52 +143,76 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
     return true;
   }
 
-  private getDropdownList = (
-    {
+  private craftResourceKey = (
+    resource: K8sResourceKind,
+    props: ResourceDropdownProps,
+  ): { customKey: any; key: any } => {
```

Comment:
nit: it will be good if you can add the types here

Reply:
Added proper types here.